### ISSUE TITLE
Appmenu support

### ISSUE THREAD

**Issue by dotWAVE** [hardpixel/unite-shell#67]
Something missing that would bring this closer to the Unity experience is appmenu. In Unity you would get entries on your top panel like "File" "Edit" "View" etc, and they would be hidden from the normal program window.

**Comment by TROMsite** [comment]
Yes! This is what you are looking for https://extensions.gnome.org/extension/1250/gnome-global-application-menu/ - unfortunately it is not developed anymore and does not work with gnome 3.30. Maybe the people behind this awesome unity-shell extension can use it and make it work with their extension. Would be fantastic!

**Comment by jmaris** [comment]
I fear the issue with this would be performance, as gnome shell performs poorly with large complex menus

**Comment by tiotrom** [comment]
1 year later and I still miss this...it was one of the most useful features in Linux...

**Comment by dotWAVE** [comment]
Seems like gnome is trying to move away from menus like this, however there are many applications that still use them.

**Comment by ripefig** [comment]
@jmaris @jonian Instead of implementing "global menu" as an actual menubar in the panel — with all the performance problems that entails — why not do it as an on-demand popup, triggered by a shortcut and by clicking a "Menu" button in the panel? You could just display all the menu items in the GTK popup, and moreover they could be made searchable. That way, you have much more flexibility and aren't dependent on the shell UI.

This would also finally resolve the terrible inconsistency between Gnome apps, which lack a menu bar, and all the other apps which still have a menubars, since in both cases all the menu items would be accessible via the same popup. For Gnome apps, additional functions could exposed via https://github.com/p-e-w/plotinus.

**Comment by otavioschwanck** [comment]
it will be a killer feature to this extension, i really want to see a global menu here

**Comment by JoveYu** [comment]
there another extension `https://github.com/gonzaarcr/Fildem` using `appmenu-gtk-module` to show global menu

**Comment by jonian** [comment]
Hi @JoveYu. I know about Fildem, it is a fork of our project https://github.com/hardpixel/gnome-hud, so there is no lack of knowledge on how to implement this feature. 

The issue is marked as `help wanted` because it needs a lot of work and I do not have currently the free time to do it. Also there are other features that are more important like #82.

**Comment by gnat** [comment]
> there another extension `https://github.com/gonzaarcr/Fildem` using `appmenu-gtk-module` to show global menu

Frustratingly, although Fildem's fork does work, it suffers from two major issues:

1. Only works on single-monitor setups. :frowning_face: https://github.com/gonzaarcr/Fildem/issues/136
2. The mouse behavior is extremely buggy: https://github.com/gonzaarcr/Fildem/issues/151

I bet @jonian and the wonderful hard-pixel folks could do this right. I could take a stab at it but don't really have the in-depth knowledge to do so. Currently using [Ubuntu Unity](https://gitlab.com/ubuntu-unity/ubuntu-remixes/ubuntu-unity/-/blob/jammy/build/switch.sh) as a substitute.

Any other help or suggestions appreciated.

**Comment by carlossantos74** [comment]
@gnat Is it possible to use ubuntu unix with gnome 40 for global menu only?

**Comment by jonian** [comment]
@tio-trom Sorry there is no way to unsubscribe you from the issue. Only you can do it from your account.

**Comment by Fuseteam** [comment]
looks like fildem got forked again https://github.com/Weather-OS/Fildem-v2/issues/14

**Comment by igor-kobzev** [comment]
It got forked and promptly abandoned again.
What a shame. It's such a useful extension, I don't understand how is it nobody wants to maintain it :(
